hudi-spark/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -113,9 +113,6 @@ class MergeOnReadSnapshotRelation(val sqlContext: SQLContext,
       hadoopConf = sqlContext.sparkSession.sessionState.newHadoopConf()
     )
 
-    // Follow the implementation of Spark internal HadoopRDD to handle the broadcast configuration.
```

Comment:
is this change needed for this PR>. some context on why this was needed?

Reply:
```SparkHadoopUtil``` becomes ```private``` in spark 3, so I need to see if these lines are necessary.

The reason this is needed in internal implementation is to guard against case where user passes a custom Configuration which doesn't contain credentials to access secure HDFS (https://github.com/apache/spark/pull/2676). Since the Configuration being used here was created as part of spark context the credentials should already be loaded. So we can remove it.